Sources/SwiftSyntax/SourceLocation.swift

```diff
@@ -104,11 +164,31 @@ public final class SourceLocationConverter {
     self.source = tree.syntaxTextBytes
     (self.lines, endOfFile) = computeLines(tree: Syntax(tree))
     precondition(tree.byteSize == endOfFile.utf8Offset)
+
+    for directive in SourceLocationCollector.collectSourceLocations(in: tree) {
+      let location = self.physicalLocation(for: directive.positionAfterSkippingLeadingTrivia)
+      if let args = directive.args {
+        let logicalFile = args.fileName.segments.description
+        guard let logicalLine = Int(args.lineNumber.description) else {
+          // FIXME: Handle hex line numbers
```

Comment:
I'm not convinced we care about this case.

Reply:
I added a doc comment saying that we ignore invalid `#sourceLocation` literals.